tests/v2/validation/nodescaling/scaling_node_driver_k3s_test.go

```diff
@@ -0,0 +1,116 @@
+	"github.com/rancher/rancher/tests/framework/extensions/scalinginput"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	password "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type K3SNodeScalingTestSuite struct {
+	suite.Suite
+	client        *rancher.Client
+	session       *session.Session
+	scalingConfig *scalinginput.Config
+}
+
+func (s *K3SNodeScalingTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *K3SNodeScalingTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.scalingConfig = new(scalinginput.Config)
+	config.LoadConfig(scalinginput.ConfigurationFileKey, s.scalingConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(s.T(), err)
+
+	newUser.Password = user.Password
+}
+
+func (s *K3SNodeScalingTestSuite) TestScalingK3SNodePools() {
+	nodeRolesEtcd := machinepools.NodeRoles{
+		ControlPlane: true,
+		Quantity:     1,
+	}
+
+	nodeRolesControlPlane := machinepools.NodeRoles{
+		Etcd:     true,
+		Quantity: 1,
+	}
+
+	nodeRolesWorker := machinepools.NodeRoles{
+		Worker:   true,
+		Quantity: 1,
+	}
+
+	nodeRolesTwoWorkers := machinepools.NodeRoles{
+		Worker:   true,
+		Quantity: 2,
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles machinepools.NodeRoles
+		client    *rancher.Client
+	}{
+		{"Scaling control plane machine pool by 1", nodeRolesEtcd, s.client},
+		{"Scaling etcd node machine pool by 1", nodeRolesControlPlane, s.client},
+		{"Scaling worker node machine pool by 1", nodeRolesWorker, s.client},
+		{"Scaling worker node machine pool by 2", nodeRolesTwoWorkers, s.client},
+	}
+
+	for _, tt := range tests {
+		clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
+		require.NoError(s.T(), err)
+
+		s.Run(tt.name, func() {
+			ScalingRKE2K3SNodePools(s.T(), s.client, clusterID, tt.nodeRoles)
+		})
+	}
+}
+
+func (s *K3SNodeScalingTestSuite) TestScalingK3SNodePoolsDynamicInput() {
+	if s.scalingConfig.NodesAndRoles == nil {
+		s.T().Skip()
+	}
+
+	clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
+	require.NoError(s.T(), err)
+
+	ScalingRKE2K3SNodePools(s.T(), s.client, clusterID, *s.scalingConfig.NodesAndRoles)
```

Comment:
since you have the static runs in an s.Run, do you want dynamic like that too? same for all the test files

Reply:
I was following how provisioning package does it and it doesn't follow that in the dynamic tests.